src/indra_cogex/analysis/source_targets_explanation.py

```diff
@@ -313,249 +311,299 @@ def get_go_terms_for_source(source_hgnc_id):
     """
-    reactome_df = pd.read_csv(reactome_filename)
-    wikipathways_df = pd.read_csv(wiki_filename)
-    pathways_df = pd.concat([reactome_df, wikipathways_df])
+    # Get all gene IDs
+    all_genes = [source_id] + target_ids
+
+    # Get pathway data using database calls
+    reactome_df = reactome_ora(
+        client=client,
+        gene_ids=all_genes
+    )
+
+    wiki_df = wikipathways_ora(
+        client=client,
+        gene_ids=all_genes
+    )
+
+    # Combine results
+    pathways_df = pd.concat([reactome_df, wiki_df])
 
     return pathways_df
 
 
-def graph_boxplots(shared_go_df, shared_entities, filename):
-    """Create boxplots to visualize p and q values
+def graph_boxplots(shared_go_df, shared_entities):
+    """Create boxplots to visualize p and q values.
 
     Parameters
     ----------
-    shared_complexes_df : dataframe
-        Contains shared bioentities that have the same go terms
-        between the GO terms provided from the gene analysis and GO terms
-        associated with source protein.
-    shared_entities : dataframe
-        The filtered the indra_upstream_df using the shared_protiens list
-        (you can pick whether you want to filter the indra_upstream_df or
-        protein_df which contains all bioentities that source protein has a
-        direct INDRA relationship with).
-    filename : string
-        name of the file chart will be downloaded under
-    """
+    shared_go_df : pd.DataFrame
+        DataFrame with GO terms analysis
+    shared_entities : pd.DataFrame
+        DataFrame with bioentities analysis
 
-    # plots boxplots for each type of graph
+    Returns
+    -------
+    :
+        Base64 encoded string of the plot
+    """
     fig, axs = plt.subplots(2, 2, figsize=(12, 8))
 
-    axs[0, 0].set_title("P-values for Shared Go Terms")
-    shared_go_df.boxplot(column=["p-value"], ax=axs[0, 0])
+    if not shared_go_df.empty:
+        axs[0, 0].set_title("P-values for Shared Go Terms")
+        shared_go_df.boxplot(column=["p"], ax=axs[0, 0])
+
+        axs[0, 1].set_title("Q-values for Shared Go Terms")
+        shared_go_df.boxplot(column=["q"], ax=axs[0, 1])
+
+    if not shared_entities.empty:
+        axs[1, 0].set_title("P-values for Shared Bioentities")
+        shared_entities.boxplot(column=["p"], ax=axs[1, 0])
+
+        axs[1, 1].set_title("Q-values for Shared Bioentities")
+        shared_entities.boxplot(column=["q"], ax=axs[1, 1])
+
+    plt.tight_layout()
 
-    axs[0, 1].set_title("Q-values for Shared Go Terms")
-    shared_go_df.boxplot(column=["q-value"], ax=axs[0, 1])
+    # Use temporary file instead of BytesIO
+    from tempfile import NamedTemporaryFile
+    with NamedTemporaryFile(suffix='.png') as tmpfile:
+        fig.savefig(tmpfile.name, format='png', bbox_inches='tight')
+        with open(tmpfile.name, 'rb') as f:
+            plot_data = base64.b64encode(f.read()).decode('utf-8')
```

Comment:
It shouls be possible to save straight to a Base64 string: https://stackoverflow.com/questions/38061267/matplotlib-graphic-image-to-base64